src/test/java/bio/terra/service/snapshot/SnapshotServiceTest.java

```diff
@@ -1203,6 +1207,50 @@ void makeSnapshotFromSnapshotRequestByRequestId() {
     assertThat(actual.getCreationInformation(), is(contentsModel));
   }
 
+  @Test
+  void getSourceDatasetsFromSnapshotRequestHandlesByRequestId() {
+    UUID snapshotAccessRequestId = UUID.randomUUID();
+    SnapshotRequestModel snapshotRequestModel = new SnapshotRequestModel();
+    SnapshotRequestContentsModel contentsModel = new SnapshotRequestContentsModel();
+    contentsModel.datasetName("datasetName");
```

Comment:
If you're not using a fluent constructor, I think it's clearer to use the `set` versions of these APIs
```suggestion
    contentsModel.setDatasetName("datasetName");
```